core/statistics/machine/cpuStatistics.go

```diff
@@ -11,33 +11,36 @@ var durationSecond = time.Second
 
 // CpuStatistics can compute the cpu usage percent
 type CpuStatistics struct {
+	numCpu          int
 	cpuPercentUsage uint64
 }
 
+// NewCpuStatistics will create a CpuStatistics object
+func NewCpuStatistics() *CpuStatistics {
+	numCpu, _ := cpu.Counts(true)
+
+	return &CpuStatistics{
+		cpuPercentUsage: 0,
+		numCpu:          numCpu,
+	}
+}
+
 // ComputeStatistics computes the current cpu usage. It should be called on a go routine as it is a blocking
 // call for a bounded time (1 second)
 func (cs *CpuStatistics) ComputeStatistics() {
 	currentProcess, err := GetCurrentProcess()
 	if err != nil {
-		cs.setZeroStatsAndWait()
 		return
 	}
 
-	cpuUsagePercent, err := currentProcess.CPUPercent()
+	percent, err := currentProcess.Percent(time.Second)
 	if err != nil {
-		cs.setZeroStatsAndWait()
 		return
 	}
 
-	numPhysicalCores, err := cpu.Counts(false)
-	if err != nil {
-		cs.setZeroStatsAndWait()
-		return
-	}
+	cpuUsagePercent := percent / float64(cs.numCpu)
 
-	cpuUsagePercent = cpuUsagePercent / float64(numPhysicalCores)
 	atomic.StoreUint64(&cs.cpuPercentUsage, uint64(cpuUsagePercent))
-	time.Sleep(durationSecond)
 }
 
 func (cs *CpuStatistics) setZeroStatsAndWait() {
```

Comment:
Is there this method used anymore? If not, you can remove it

Reply:
Done. Method removed.